pkg/snapshot/controllers/snapshotrestore.go

```diff
@@ -155,6 +157,12 @@ func (c *SnapshotRestoreController) handleInitial(snapRestore *stork_api.VolumeS
 			return fmt.Errorf("unable to get get snapshot  details %s: %v",
 				snapName, err)
 		}
+		if err := k8s.Instance().ValidateSnapshot(snapName,
```

Comment:
We shouldn't retry here. Just check if the snapshot is bound. If not raise an event and return nil, it'll be retried in the next reconcile loop

Reply:
bound meaning whether spec.SnapshotDataName is not empty ?

Reply:
Set the 3rd param for `ValidateSnaphshot()` to false to not retry

Reply:
set `retry` to false